notebooks/official/ray_on_vertex_ai/spark_on_ray_on_vertex_ai.ipynb

```diff
@@ -0,0 +1,1470 @@
+        "Ray [Task](https://docs.ray.io/en/latest/ray-core/tasks.html#ray-remote-functions) or [Actor](https://docs.ray.io/en/latest/ray-core/actors.html), is required for creating a Spark session with [Ray client](https://docs.ray.io/en/latest/cluster/running-applications/job-submission/ray-client.html) on the Ray on Vertex AI.  The following code shows how a Ray Actor can be used for creating a Spark Session, running a Spark application, and stopping a Spark session on a Ray on Vertex AI using RayDP.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f59c07ce1d52"
+      },
+      "outputs": [],
+      "source": [
+        "import ray\n",
+        "\n",
+        "\n",
+        "@ray.remote\n",
+        "class SparkExecutor:\n",
+        "    import pyspark\n",
+        "\n",
+        "    spark: pyspark.sql.SparkSession = None\n",
+        "\n",
+        "    def __init__(self):\n",
+        "\n",
+        "        import raydp\n",
+        "\n",
+        "        self.spark = raydp.init_spark(\n",
+        "            app_name=\"RAYDP ACTOR EXAMPLE\",\n",
+        "            num_executors=1,\n",
+        "            executor_cores=1,\n",
+        "            executor_memory=\"500M\",\n",
+        "        )\n",
+        "\n",
+        "    def get_data(self):\n",
+        "        df = self.spark.createDataFrame(\n",
+        "            [\n",
+        "                (\"sue\", 32),\n",
+        "                (\"li\", 3),\n",
+        "                (\"bob\", 75),\n",
+        "                (\"heo\", 13),\n",
+        "            ],\n",
+        "            [\"first_name\", \"age\"],\n",
+        "        )\n",
+        "        return df.toJSON().collect()\n",
+        "\n",
+        "    def stop_spark(self):\n",
+        "        import raydp\n",
+        "\n",
+        "        raydp.stop_spark()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "037540acbcca"
+      },
+      "source": [
+        "### Connect to Ray cluster on Vertex AI"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "b31d9281a41d"
+      },
+      "outputs": [],
+      "source": [
+        "RAY_ADDRESS = f\"vertex_ray://{ray_cluster_resource_name}\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5303ede32b8e"
+      },
+      "source": [
+        "Ray client may time out if we try to connect immediately, so add a wait of 5 minutes"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "e6a42d72e2b3"
+      },
+      "outputs": [],
+      "source": [
+        "import time\n",
+        "\n",
+        "for i in range(0, 300):\n",
+        "    time.sleep(1)\n",
+        "    print(f\"Waited for total {i} seconds...\")"
```

Comment:
I understand for testing purposes that we need to have this timeout here, but is there a cleaner way to implement it? Is it possible to retry the init call in the next cell multiple times if there's a failure?